reader/reader_test.go

```diff
@@ -0,0 +1,36 @@
+package reader
+
+import (
+	"log"
+	"os"
+	"testing"
+)
+
+func TestReadCsv(t *testing.T) {
+	name := "test.csv"
+	f := createFile(name)
+
+	dat := ReadCsv(name)
```

Comment:
Notice that we are passing a name to ReadCsv (the name should be ReadCSV BTW). It is better to pass the file coming back from `createFile` function I have refactored, which you can do all sorts of things with it. When the project grows, it is better not to create these kind of objects all the time and pass them around.